src/operations.js

```diff
@@ -14,13 +14,27 @@ const combinePreviews = (p1, p2) => (x) => {
 }
 
 /**
- * iso/prism/lens/affinetraversal/getter/affinefold/traversal/reviewer/fold/setter
+ * Compose two optics
  *
  * @param {*} optic1
  * @param {*} optic2
  */
 const compose2Optics = (optic1, optic2) => {
-  // start from most specific (iso) to less specific (fold, setter, review)
+  // turn strings and numbers into proper optics
+  if (typeof optic1 == 'string' || optic1 instanceof String) {
+    return compose2Optics(optionalProp(optic1), optic2)
+  }
+  if (typeof optic1 == 'number') {
```

Comment:
Maybe we could also make sure here the value is not `NaN` (stupid JS behaviour) 🤦 
```suggestion
  if (typeof optic1 == 'number' || !isNaN(optic1) {
```
https://flaviocopes.com/how-to-check-value-is-number-javascript/

Reply:
Done in the latest commits